Android/app/src/main/java/com/example/androidgui/inventoryactivities/UpdateQuanActivity.java

```diff
@@ -8,13 +8,17 @@
 import android.os.Bundle;
 import com.example.androidgui.R;
 import presenter.inventorysystem.InventoryManager;
+import presenter.inventorysystem.InventoryPresenter;
+import presenter.inventorysystem.UpdateQuanInterface;
 import use_case.boundary.output.InventoryOutputBoundary;
 
-public class UpdateQuanActivity extends AppCompatActivity implements View.OnClickListener, InventoryOutputBoundary {
+public class UpdateQuanActivity extends AppCompatActivity implements View.OnClickListener, InventoryOutputBoundary,
+        UpdateQuanInterface{
     Button btn;
     EditText name;
     EditText usage;
     final InventoryManager im = new InventoryManager();
+    InventoryPresenter ip = new InventoryPresenter();
```

Comment:
are all of these activities using the same presenter? Generally, you want one presenter per activity otherwise errors can easily happen